data/transactions/logic/fields.go

```diff
@@ -474,50 +549,62 @@ const (
 	Secp256k1 EcdsaCurve = iota
 	// Secp256r1 curve
 	Secp256r1
-	invalidEcdsaCurve
+	invalidEcdsaCurve // compile-time constant for number of fields
 )
 
-// EcdsaCurveNames are arguments to the 'ecdsa_' opcode
-var EcdsaCurveNames []string
+var ecdsaCurveNames [invalidEcdsaCurve]string
 
 type ecdsaCurveSpec struct {
 	field   EcdsaCurve
 	version uint64
+	doc     string
 }
 
-func (fs *ecdsaCurveSpec) Type() StackType {
+func (fs ecdsaCurveSpec) Field() byte {
+	return byte(fs.field)
+}
+
+func (fs ecdsaCurveSpec) Type() StackType {
 	return StackNone // Will not show, since all are the same
 }
 
-func (fs *ecdsaCurveSpec) OpVersion() uint64 {
+func (fs ecdsaCurveSpec) OpVersion() uint64 {
 	return 5
 }
 
-func (fs *ecdsaCurveSpec) Version() uint64 {
+func (fs ecdsaCurveSpec) Version() uint64 {
 	return fs.version
 }
 
-func (fs *ecdsaCurveSpec) Note() string {
-	note := EcdsaCurveDocs[fs.field.String()]
-	return note
+func (fs ecdsaCurveSpec) Note() string {
+	return fs.doc
 }
 
-var ecdsaCurveSpecs = []ecdsaCurveSpec{
-	{Secp256k1, 5},
-	{Secp256r1, fidoVersion},
+var ecdsaCurveSpecs = [...]ecdsaCurveSpec{
+	{Secp256k1, 5, "secp256k1 curve, used in Bitcoin"},
+	{Secp256r1, fidoVersion, "secp256r1 curve, NIST standard"},
 }
 
-var ecdsaCurveSpecByField map[EcdsaCurve]ecdsaCurveSpec
+func ecdsaCurveSpecByField(c EcdsaCurve) (ecdsaCurveSpec, bool) {
+	if int(c) >= len(ecdsaCurveSpecs) {
+		return ecdsaCurveSpec{}, false
+	}
+	return ecdsaCurveSpecs[c], true
+}
 
-// EcdsaCurveSpecByName gives access to the field specs by field name
-var EcdsaCurveSpecByName ecDsaCurveNameSpecMap
+var ecdsaCurveSpecByName = make(ecDsaCurveNameSpecMap, len(ecdsaCurveNames))
 
-// simple interface used by doc generator for fields versioning
 type ecDsaCurveNameSpecMap map[string]ecdsaCurveSpec
```

Comment:
@jannotti Minor:  is _D_ capitalized intentionally?  It feels like the intended reading is `ecdsaCurveNameSpecMap`.

Reply:
Nope, changing.